framework/physics/physics_material/multi_group_xs/single_state_mgxs.cc

```diff
@@ -110,37 +108,38 @@ SingleStateMGXS::MakeCombined(std::vector<std::pair<int, double>>& combinations)
 {
   Clear();
 
-  // pickup all xs and make sure valid
+  // Pickup all xs and make sure valid
   std::vector<std::shared_ptr<MultiGroupXS>> xsecs;
   xsecs.reserve(combinations.size());
 
   unsigned int n_grps = 0;
   unsigned int n_precs = 0;
-  double Nf_total = 0.0; // total density of fissile materials
+  double Nf_total = 0.0; // Total density of fissile materials
 
-  // loop over cross sections
+  // Loop over cross sections
   for (auto combo : combinations)
   {
-    // get the cross section from the lua stack
+    // Get the cross section from the stack
     std::shared_ptr<MultiGroupXS> xs;
     xs = Chi::GetStackItemPtr(Chi::multigroup_xs_stack, combo.first, std::string(__FUNCTION__));
     xsecs.push_back(xs);
 
-    // increment densities
+    // Set the scaling factor
+    std::dynamic_pointer_cast<SingleStateMGXS>(xs)->SetScalingFactor(combo.second);
```

Comment:
curious: why do we cast combo.second now?

Reply:
We are not casting `combo.second` we are casting `xs` here. The `MultiGroupXS` class is abstract and does not have many of the methods, so here we cast to the concrete class `SingleStateMGXS` where `SetScalingFactor` is implemented. I am open to discussions on reworking the cross sections.

Reply:
ok. @andrsd what is your opinion of l.128? I am ignorant of such matters.

Reply:
If your concern is about `dynamic_cast`, then they exist for a reason. But for example, you do not want to do that inside a quadrature point loop.

So, this is ok, unless I missed something. If refactoring happens in cross-section classes, this can be changed then. No need to stress about this...